library/Freckle/App/Stats.hs

```diff
@@ -108,17 +133,26 @@ withStatsClient
   -> (StatsClient -> m a)
   -> m a
 withStatsClient StatsSettings {..} f = do
+  gauges <- liftIO $ fmap Gauges EKG.new
   if amsEnabled
     then do
       tags <- (amsTags <>) <$> getEcsMetadataTags
-
       Datadog.withDogStatsD amsSettings $ \client ->
         -- Add the tags to the thread context so they're present in all logs
         withThreadContext (map toPair tags)
-          $ f StatsClient { scClient = client, scTags = tags }
-    else f $ StatsClient { scClient = Datadog.Dummy, scTags = amsTags }
+          $ f StatsClient { scClient = client, scTags = tags, scGauges = gauges }
+    else do
+      f $ StatsClient { scClient = Datadog.Dummy, scTags = amsTags, scGauges = gauges }
   where toPair = bimap (fromString . unpack) String
 
+withGauge :: (MonadReader app m, HasStatsClient app, MonadUnliftIO m) => GaugeName -> m a -> m a
+withGauge gaugeName f = do
+  gauge' <- asks $ view $ statsClientL . gaugesL . to (unGaugeName gaugeName)
```

Comment:
FYI, `MonadReader` it itself a getter, so you don't need to `asks` a `view`, you can just `view`. Kind of like `MonadState` being a full lens (get and set). Pretty neat.

```suggestion
  gauge' <- view $ statsClientL . gaugesL . to (unGaugeName gaugeName)
```